_docs/setup/kubernetes/multicluster-install.md

```diff
@@ -0,0 +1,161 @@
+---
+title: Istio Multicluster
+overview: Install Istio with multicluster support.
+
+order: 65
+
+layout: docs
+type: markdown
+---
+
+{% include home.html %}
+
+Instructions for the installation of Istio multicluster.
+
+## Prerequisites
+
+* Individual cluster Pod CIDR ranges and service CIDR ranges must be unique
```

Comment:
Is this absolutely a hard requirement?  I don't think this works for IBM Cloud.

Reply:
I believe this is the case unless there is some advanced functionality added to Kubernetes in the future.  The main reason is when a multicluster mesh is instantiated, in order for deployments and their services not to collide (as one big mesh) the service IPs must not be allocated twice.  Something must do service CIDR allocation.  I could imagine an extra Kubernetes service in the future managing the service cidr allocations so multiple clusters could co-exist and use the same service CiDR.  The simple answer is simply to just force pod and service cidr allocations to always be unique.

Since routing is required between all nodes in the cluster for the PODs IP CIDRs to communicate with one another, a similar situation exists where no collision is a hard requirement.  Again something more advanced and centralized in Kubernetes could handle this.

I could also see a future where some type of egress/ingress gateway between multicluster isolated the pod and service CIDRs providing a true multicloud experience.

This document describes  the current merged state of the implementation.  I think it is important to get feedback from Operators (such as IBM Cloud) to determine the pain points around multicluster/multicloud, so those pain points can be prioritized, and those pain points can be resolved.

Without documentation/CI, the current state of the implementation cannot be understood nor used by anyone, so there would be no feedback loop as per Agile methods.

Reply:
@tiswanso can you validate the above statement I made please.

Reply:
This is the simplest way to do it,  although there are other possibilities.  AFAIK no one has attempted any other networking setups so we definitely don't want to be recommending something else at the current time.  A couple things to add to the discussion.  The current solution requires a VPN (or a single RFC-1918 compliant network).   Note when the Zero VPN work is available some of it will apply to this solution as well.  
Considering that this is a single RFC-1918 compliant network implies that the CIDRs must not over-lap.  It also implies that there needs to be Ip reachability across the clusters.  Now it would be possible to set up your app deployment and Istio deployment so service IPs that are externally reachable are used for all endpoints and services that have to reach each other across the clusters.   Using the recently added gateway functionality may also make this easier to avoid exposing the Pod IPs across the clusters.  As I noted this hasn't been tested so in may not work in practice.  Additionally it is likely too cumbersome for anyone to want to deply it that way.

Reply:
I agree, we should explain why these pre-requisites exist and probably update the docs when the gateway functionality exists as that provides a way to easily expose services across clusters.

> Now it would be possible to set up your app deployment and Istio deployment so service IPs that are externally reachable are used for all endpoints and services that have to reach each other across the clusters.

I think this is also a good option too as users could expose all the services to be reachable within user's private network.  We have such an option in ibm cloud and could be leveraged.

Reply:
@linsun I am not opposed to explaining the why of the prereqs, although it would be a strong delta from patterns in all of our other docs which don't explain why.  I am not keen to rework all the documentation in the repo to answer why questions (many that I don't have the answer to).

Reply:
@linsun I've thought this over a bit, and will followup with a diagram that takes a look at explaining why/how for many of these prereqs.  I'd prefer to do that as a separate PR, as it seems tricky and I don't want to set a standard that all documentation must answer why/how.  At present, the docs generally answer what, and I think that is a pretty good approach to managing documentation.

Related to the request to updating docs with the gateway functionality, I've had a look at some of the work taking place there.  I think it looks very promising for providing a true multicloud experience and will of course commit to updating the docs when that work is integrated with multicluster/multicloud.